sim/scripts/create_mjcf.py

```diff
@@ -170,42 +182,65 @@ def adapt_world(
             1,
             mjcf.Option(
                 timestep=0.001,
-                viscosity=1e-6,
                 iterations=50,
                 solver="PGS",
                 gravity=(0, 0, -9.81),
-                flag=mjcf.Flag(frictionloss="enable"),
             ).to_xml(),
         )
 
         visual_geom = ET.Element("default", {"class": "visualgeom"})
-        geom_attributes = {"material": "visualgeom", "condim": "1", "contype": "1", "conaffinity": "0"}
+        geom_attributes = {"material": "visualgeom", "condim": "1", "contype": "0", "conaffinity": "0"}
         ET.SubElement(visual_geom, "geom", geom_attributes)
 
         root.insert(
             1,
             mjcf.Default(
-                joint=mjcf.Joint(armature=0.01, damping=0.1, limited=True, frictionloss=0.01),
+                joint=mjcf.Joint(armature=0.01, damping=0.01, limited=True, frictionloss=0.01),
                 motor=mjcf.Motor(ctrllimited=True),
                 equality=mjcf.Equality(solref=(0.001, 2)),
                 geom=mjcf.Geom(
                     solref=(0.001, 2),
                     friction=(0.9, 0.2, 0.2),
-                    condim=3,
+                    condim=4,
                     contype=1,
-                    conaffinity=0,
+                    conaffinity=15,
                 ),
                 visual_geom=visual_geom,
             ).to_xml(),
         )
 
-        if add_reference_position:
-            root = self.add_reference_position(root)
+        # Locate actual root body inside of worldbody
+        root_body = worldbody.find(".//body")
+        # Make position and orientation of the root body
+        root_body.set("pos", "0 0 " + ROOT_HEIGHT)
+        root_body.set("quat", "1 0 0 0")
+
+        # Add cameras and imu
+        root_body.insert(0, ET.Element("camera", name="front", pos="0 -3 1", xyaxes="1 0 0 0 1 2", mode="trackcom"))
+        root_body.insert(
+            1,
+            ET.Element(
+                "camera",
+                name="side",
+                pos="-2.893 -1.330 0.757",
+                xyaxes="0.405 -0.914 0.000 0.419 0.186 0.889",
+                mode="trackcom",
+            ),
+        )
+        root_body.insert(2, ET.Element("site", name="imu", size="0.01", pos="0 0 0"))
 
-        # Move gathered elements to the new root body
-        for item in items_to_move:
-            worldbody.remove(item)
-            new_root_body.append(item)
+        # # Add joints to all the movement of the base
+        # Define the joints as individual elements
+        joints = [
```

Comment:
use free joint since it's a default mujoco logic

Reply:
addressed